### DISCUSSION TITLE
Authentication does not work in Desktop mode (macos)

### DISCUSSION THREAD

**Discussion by dkzkv** [flet-dev/flet#3170] [Q&A]
✅ Checked Issues

-    https://github.com/flet-dev/flet/issues/2936 similar issue, but its still open

✅ Checked Stackoverflow

### ℹ️ Description
Authentication in desktop mode does not work. 

### 🤖 Code example to reproduce the issue:

```python
import flet as ft
from urllib.parse import urlparse

from flet_runtime.auth.providers import GoogleOAuthProvider

GOOGLE_CLIENT_ID = "{YOUR ID HERE}"
GOOGLE_CLIENT_SECRET = "{YOUR SECRET HERE}"

async def main(page: ft.Page):
    google_provider = GoogleOAuthProvider(
        client_id=GOOGLE_CLIENT_ID,
        client_secret=GOOGLE_CLIENT_SECRET,
        redirect_url=f"http://localhost:{urlparse(page.url).port}/oauth_callback",
    )

    info_text = ft.TextField("", multiline=True)

    async def google_logon(e):
        await page.login_async(google_provider)

    async def on_login(e):
        print("Login error:", e.error)
        print("Access token:", page.auth.token.access_token)
        print("User ID:", page.auth.user.id)
        info_text.value = page.auth.user.id
        await info_text.update_async()

    page.on_login = on_login
    page.add(ft.ElevatedButton("Continue with Google", on_click=google_logon), info_text)

ft.app(target=main)
```

### ⚠️ Actual result:
When calling the function ```await page.login_async(google_provider)```, I get the following error:
```
Exception: oauthAuthorize command is not supported.
```

### 🥳 Expected result:
The function should open a window with google authentication in the browser ([docs](https://flet.dev/docs/cookbook/authentication/))

#### Attempt to fix 1
if i add ```ft.FLET_APP_WEB```:
```
ft.app(target=main, view=ft.FLET_APP_WEB)
```
Then the authentication works, it redirects me to the browser and after authentication redirects me back to the application.
**BUT**: but it only works if I run the application from the IDE. If I try to build an application 'flet build macos' and then authenticate...i get same error ```Exception: oauthAuthorize command is not supported.```.

#### Attempt to fix 2
Building ```flet build macos``` with
``` 
ft.app(target=main, port=8550, view=ft.WEB_BROWSER)
```
this leads to the fact that after the start of the application, an empty screen is displayed and the controllers are not loaded (apparently some kind of internal logic)

### Flet version::

flet=0.22.0
Python version 3.11.6
Operating system:

Macos M2
Sonoma 14.4.1
Build command:
flet build macos

**Comment by FeodorFitsner** [comment]
It's actually "as designed". Using OAuth doesn't make sense with a standalone desktop app as secret key comes with the app and any user could retrieve it.

What could work though is Flet desktop app that works with a remote web server keeping secret keys and controlling auth process.

What is your deployment scenario here?

**Reply by dkzkv** [reply]
I wrote this issue because I read the documentation and it clearly states that authentication should work with a desktop application😇

https://flet.dev/docs/cookbook/authentication/
```
Desktop app - a new browser window or tab.
```

#### How it should be:
1. The user downloads the application
2. Opens the application and sees the authorization screen, logs in via SSO (Google Auth)
3. Then after logging in, he receives information related to the profile from the server

Is there any documentation or is it planned to be how to implement authentication in desktop mode? Or Best Practices how to do it with flet framework?

**Reply by zox47** [reply]
Hello, I am having the same problem, even though I have Calendar, the error is thrown by not having "port=8550, view=flet.WEB_BROWSER" in the app(), but if we put that the app on the mobile does not work it remains blank.
